cachito/workers/tasks/npm.py

```diff
@@ -43,102 +88,128 @@ def cleanup_npm_request(request_id):
 
 
 @app.task
-def fetch_npm_source(request_id):
+def fetch_npm_source(request_id, package_configs=None):
     """
     Resolve and fetch npm dependencies for a given request.
 
+    This function uses the Python ``os.path`` library to manipulate paths, so the path to the
+    configuration files may differ in format based on the system the Cachito worker is deployed on
+    (i.e. Linux vs Windows).
+
     :param int request_id: the Cachito request ID this is for
+    :param list package_configs: the list of optional package configurations submitted by the user
     :raise CachitoError: if the task fails
     """
+    if package_configs is None:
+        package_configs = []
+
     validate_npm_config()
 
     bundle_dir = RequestBundleDir(request_id)
     log.debug("Checking if the application source uses npm")
-    for lock_file in (bundle_dir.npm_shrinkwrap_file, bundle_dir.npm_package_lock_file):
-        if lock_file.exists():
-            break
-    else:
-        raise CachitoError(
-            "The npm-shrinkwrap.json or package-lock.json file must be present for the npm "
-            "package manager"
-        )
+    subpaths = []
+    if package_configs:
```

Comment:
`package_configs` is set to an empty list if it's `None`. We could completely remove this if statement, and the explicit initialization of `subpaths` above..